File: Exercicio37.py
numeros = []
n=0 
valor= 0 
posição=0 
novamente ='S'

N = int(input('Digite a quantidade de valores que serão armazenados no vetor:'))

if(N<=0 or N>20):
 N = int(input('Erro! Apenas numeros positivos. E até 20:'))


for i in range(0, N, 1):
  x = int(input(f'Digite {N} numeros :'))
  numeros.append(x)

while(novamente =='S'):

      valor  = int(input('Digite um número para identificar a posição que ele ocupa no vetor: '))

      for i in range(0,N, 1):
        if (valor == numeros[i]):
            posição= i
            print (f'A posição ocupada no vetor é {posição}:')
            novamente ='G'
            break
      else:
            print ('Valor não encontrado!')

      novamente = (input('Deseja saber a posição de outro número (S-sim) e (N -não):'))

 

File: Exercicio37.c
#include <stdio.h>
 
int main(int argc, char *argv[]){
  int *numeros;
  int qtd, escolha, posicao;
  char tentativa;

  posicao=21;
  tentativa='S';

  printf("Digite a quantidade de numeros: ");
  scanf("%i", &qtd);

   while(qtd > 20){
    printf("Erro! Somente numeros menores do que 20. Digite novamente: ");
    scanf("%i", &qtd);
  }
 
  // O malloc reserva 100 blocos na memória e o ponteiro "numeros" aponta pra lá
  numeros = (int*)malloc(qtd*sizeof(int));
 
  printf("Digite os numeros: ");
  for(int i=0; i<qtd; i++){
      scanf("%i", &numeros[i]);
  }

  while (tentativa == 'S') {
    printf("Escolha um numero da sequencia digitada anteriormente: ");
    scanf("%i", &escolha);

     for(int i=0; i<qtd; i++){
       if (escolha == numeros[i]) {

           posicao=i+1;
       }     
     }
      if (posicao<20) {
      printf ("Esse numero se encontra na posicao %i", posicao);
      }
          
     else {

        printf ("Valor nao encontrado!");
     
  }
      printf ("\nDeseja pesquisar outro numero (S-Sim e N-Nao): ");
      scanf("%s", &tentativa);
     }

 printf ("\n Obrigada!");
     

    


    return 0;
}

